### PR TITLE
Add basic tests for shaded JAR

### LINKED ISSUE #12711: Test shaded jar as node/transport client
Need integ test in qa/ (shading is complicated)

### PR DESCRIPTION
**Pull request by s1monw**
this commit adds a simple integration test that starts a
node from a shaded jar, indexes a doc and retrieves it. It
also has some basic unittests that try to load shaded classes and ensure
that their counterpart is not in the classpath.

Closes #12711

### CODE REVIEW

File: qa/pom.xml [hunk 1]
```diff
@@ -33,188 +33,7 @@
             <artifactId>lucene-test-framework</artifactId>
             <scope>test</scope>
         </dependency>
-        <dependency>
```
Comment (nik9000) [inline qa/pom.xml]: +1

**Comment by nik9000** [comment]
LGTM

**Comment by s1monw** [comment]
@rmuir if u have a sec take a look

**Comment by rmuir** [comment]
+1 fantastic

**Pull request merged**